Makefile

```diff
@@ -1,58 +1,72 @@
-PKGNAME := $(shell cat package.json | jq -r .name)
+NPM_PKGNAME := $(shell cat package.json | jq -r .name)
+PYTHON_PKGNAME := mozilla_nimbus_shared
 VERSION := $(shell cat package.json | jq -r .version)
 
 # Schemas are anything we can find in the schema directory, plus the directory itself
 SCHEMAS := schemas $(shell test -d schemas && find schemas -name '*.json')
 TYPES := $(shell find ./types -name '*.ts')
 TS_SRC := index.ts $(shell find ./src -name '*.ts')
-PACK_FILE := $(PKGNAME)-$(VERSION).tgz
+PY_SRC := $(shell find ./python/mozilla_nimbus_shared -name '*.py')
+NPM_PACK_FILE := $(shell echo $(NPM_PKGNAME) | sed -e 's/@//' -e 's_/_-_')-$(VERSION).tgz
+PYTHON_PACK_FILE := python/dist/$(PYTHON_PKGNAME)-$(VERSION).tar.gz python/dist/$(PYTHON_PKGNAME)-$(VERSION)-py3-none-any.whl
 GENERATED_TS := ./src/_generated/typeGuardHelpers.ts ./src/_generated/schemas.ts
+GENERATED_PYTHON := ./python/pyproject.toml
+GENERATED := $(GENERATED_TS) $(GENERATED_PYTHON)
 
 TIMESTAMP_DIR := ./.timestamps
 TSC_STAMP := $(TIMESTAMP_DIR)/tsc-last-run
 NPM_INSTALL_STAMP := $(TIMESTAMP_DIR)/npm-install
+PYTHON_INSTALL_STAMP := $(TIMESTAMP_DIR)/poetry-install
 
 export PATH := node_modules/.bin:$(PATH)
 
-# Phony commands - commands that don't make files
-.PHONY: install build clean test lint default clean-build pack
+# Phony targets - commands that don't make files
+.PHONY: install build clean test lint default pack
 
 default: build
 
-install: $(NPM_INSTALL_STAMP)
+install: $(NPM_INSTALL_STAMP) $(PYTHON_INSTALL_STAMP)
 
-build: $(TSC_STAMP) $(SCHEMAS)
+build: $(TSC_STAMP) $(SCHEMAS) $(GENERATED)
 
 clean:
 	rm -rf dist $(SCHEMAS) $(TSC_STAMP) $(NPM_INSTALL_STAMP) node_modules
-	rm -rf $(TIMESTAMP_DIR) $(GENERATED_TS)
+	rm -rf $(TIMESTAMP_DIR) $(GENERATED)
 
 test: build
 	./test/normandy-arguments.ts
 
-artifact: build
+artifact: build pack
 	./bin/pack-artifact.sh
+	mkdir -p artifacts/python
+	mv python/dist/mozilla_nimbus_shared-$(VERSION)-py3-none-any.whl artifacts/python/
+	mv python/dist/mozilla_nimbus_shared-$(VERSION).tar.gz artifacts/python/
+	mkdir -p artifacts/npm
+	mv mozilla-rapid-experiments-shared-$(VERSION).tgz artifacts/npm/
 
 lint: $(NPM_INSTALL_STAMP) build
 	eslint .
 
-clean-build: clean build
+pack: $(NPM_PACK_FILE) $(PYTHON_PACK_FILE)
 
-pack: $(PACK_FILE)
-
-# Commands that make files
+# Commands that make files. None of these should depend on phony targets
 
 $(NPM_INSTALL_STAMP): package.json package-lock.json
 	npm ci
 	@mkdir -p $(@D)
 	@touch $@
 
+$(PYTHON_INSTALL_STAMP): python/pyproject.toml
+	cd python; poetry install
+	@mkdir -p $(TIMESTAMP_DIR)
+	@touch $(PYTHON_INSTALL_STAMP)
+
 $(TSC_STAMP): $(GENERATED_TS) $(TS_SRC) $(TYPES) $(NPM_INSTALL_STAMP) tsconfig.json
-	tsc
 	@mkdir -p $(@D)
+	tsc
 	@touch $@
 
-$(SCHEMAS): $(TYPES) $(NPM_INSTALL_STAMP) bin/build-schemas.ts
+$(SCHEMAS) &: $(TYPES) $(NPM_INSTALL_STAMP) bin/build-schemas.ts
```

Comment:
Seeing these warnings in CircleCI logs (here and on L84 below):

```
Makefile:85: warning: overriding recipe for target '&'
Makefile:70: warning: ignoring old recipe for target '&'
eslint .
```

Reply:
First off, thanks for looking through those logs. The `&:` syntax is an extension of GNU Make that allows you to say that multiple targets are all built together by the same command (as opposed to running the command once per target). I guess CI isn't using GNU Make like I am locally. That means that CI is probably doing some extra work rebuilding things, but it isn't a problem.

It shouldn't be harmful, but definitely something I'd like to fix. Lets do that in another PR though.